File: Python/167. Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        a, b = 0, len(numbers) - 1
        while numbers[a] + numbers[b] != target:
            if numbers[b] + numbers[a] > target: b -= 1
            else: a += 1
        return [a + 1, b + 1]

File: Rust/167. Two Sum II - Input Array Is Sorted.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut a = numbers.len() - 1;
        let mut b = 0;
        while numbers[a] + numbers[b] != target {
            if numbers[a] + numbers[b] > target { a -= 1; }
            else{ b += 1; }
        }
        return vec![(b + 1) as i32, (a + 1) as i32];
    }
}
